Sources/SwiftDocC/Semantics/Symbol/Symbol.swift

```diff
@@ -323,12 +332,75 @@ public final class Symbol: Semantic, Abstracted, Redirected, AutomaticTaskGroups
         self.redirectsVariants = redirectsVariants
         self.originVariants = originVariants
         self.automaticTaskGroupsVariants = automaticTaskGroupsVariants
-        self.extendedModule = extendedModule
     }
     
     public override func accept<V: SemanticVisitor>(_ visitor: inout V) -> V.Result {
         return visitor.visitSymbol(self)
     }
+
+    /**
+     Append a new generic constraint for the given extended module
+
+     - Parameters:
+        - extendedModule: Which module this symbol extends.
+        - typeKind: The ``SymbolGraph/Symbol/KindIdentifier`` of the symbol this extension extends.
+        - constraint: The new generic constraints to add
+        - trait: Which generic constraints variant to append to
+     - Returns: True if the constraint was appended to an existing or new swift extension variant. False if there was an existing swift extension variant for a different module.
+     */
+
+    public func addConstraint(
+        extendedModule: String,
+        typeKind: SymbolGraph.Symbol.KindIdentifier? = nil,
+        constraint newConstraint: SymbolGraph.Symbol.Swift.GenericConstraint,
+        trait: DocumentationDataVariantsTrait = DocumentationDataVariantsTrait.swift
+    ) {
+
+        var swiftExtension: SymbolGraph.Symbol.Swift.Extension
+
+        // Does this symbol already have a swift extension variant for the requested trait?
+        
+        // Yes: Create a new copy of the existing extension with
+        // the new constraint appended to the existing list
+        if let existing = swiftExtensionVariants()[trait] {
+            // Double check the existing extension uses the same module and type. If it does not,
+            // we must have a tooling or data consistency problem.
+            assert(
+                existing.extendedModule == extendedModule, //&& existing.typeKind == typeKind,
+                "New constraint's module and type kind do not match symbol's existing constraints."
+            )
+            swiftExtension = existing
+            swiftExtension.constraints = swiftExtension.constraints + [newConstraint]
+
+        // No: Create a new extension with the specified module, type and
+        // new constraint
+        } else {
+            swiftExtension = SymbolGraph.Symbol.Swift.Extension(
+                                extendedModule: extendedModule,
+                                typeKind: typeKind,
+                                constraints: [newConstraint]
+                             )
+        }
+
+        // Save the new or updated extension
+        self.mixinsVariants[
+                trait,
+                default: [:]
+            ][SymbolGraph.Symbol.Swift.Extension.mixinKey] = swiftExtension
+    }
+
+    // MARK: - Private helpers
+
+    /// Return all of this symbol's Swift extension variants.
+    func swiftExtensionVariants() -> [DocumentationDataVariantsTrait : SymbolGraph.Symbol.Swift.Extension] {
```

Comment:
If nothing other then `Symbol` needs to call this we could make it `private`

Reply:
Yes thank you. Marked private.